lib/rules/landmark-unique-matches.js

```diff
@@ -46,3 +51,20 @@ function isLandmarkVirtual(vNode) {
 function isHeaderFooterLandmark(headerFooterElement) {
   return !closest(headerFooterElement, excludedParentsForHeaderFooterLandmarks);
 }
+
+/*
+ * `aside` elements map to `complementary` landmarks when scoped to `body` or
+ * `main`. If it is within sectioning content, it is not a landmark unless it
+ * has an accessible name.
+ *
+ * https://w3c.github.io/html-aam/#el-aside-ancestorbodymain
+ */
+const sectioningContentSelector = sectioningContentElements.join(',');
+function isAsideLandmark(asideElement) {
+  return (
+    !closest(
+      asideElement.parent, // closest() can match self, which we do not want, so start at parent element
+      sectioningContentSelector
+    ) || !!accessibleTextVirtual(asideElement)
```

Comment:
I'm not 100% sure that this function, accessibleTextVirtual(), is 100% equivalent to the "accessible name" requirement of the specification ([Section 3.4.9](https://w3c.github.io/html-aam/#el-aside))